docs/coding-guidelines.md

```diff
@@ -171,6 +171,29 @@ For example the following are correct:
     }
     ```
 
+1. Do not use `Func<>` or `Action<>` as return types. Instead, let the compiler convert the method into a func or action as necessary.
+
+    This is correct:
```

Comment:
I like the whole 'this is correct', 'this is incorrect" separation, but it does feel repetitive. 

This is not a criticism on your PR, but the general guidelines. 

They are getting long :)